File: src/main/java/io/github/freya022/botcommands/api/commands/prefixed/TextCommandsContext.java
package io.github.freya022.botcommands.api.commands.prefixed;

import io.github.freya022.botcommands.api.core.service.annotations.InjectedService;
import io.github.freya022.botcommands.internal.commands.prefixed.TopLevelTextCommandInfo;

import java.util.Collection;

@InjectedService
public interface TextCommandsContext {
	Collection<TopLevelTextCommandInfo> getRootCommands();
}

File: src/main/java/io/github/freya022/botcommands/api/commands/prefixed/TextCommandsContext.kt
package io.github.freya022.botcommands.api.commands.prefixed

import io.github.freya022.botcommands.api.core.service.annotations.InjectedService
import io.github.freya022.botcommands.internal.commands.prefixed.TextCommandInfo
import io.github.freya022.botcommands.internal.commands.prefixed.TopLevelTextCommandInfo

/**
 * Helps to get the registered text commands.
 */
@InjectedService
interface TextCommandsContext {
    val rootCommands: Collection<TopLevelTextCommandInfo>

    fun findTextCommand(words: List<String>): TextCommandInfo?

    fun findTextSubcommands(words: List<String>): Collection<TextCommandInfo>
}